lib/ansible/modules/packaging/language/pip.py

```diff
@@ -234,6 +260,94 @@
 _SPECIAL_PACKAGE_CHECKERS = {'setuptools': 'import setuptools; print(setuptools.__version__)',
                              'pip': 'import pkg_resources; print(pkg_resources.get_distribution("pip").version)'}
 
+op_dict = {">=": operator.ge, "<=": operator.le, ">": operator.gt,
+           "<": operator.lt, "==": operator.eq, "!=": operator.ne}
+
+
+class Distribution:
+    """Python distribution package metadata wrapper.
+
+    A wrapper class for Requirement, which provides API to parse package name
+    and version specifier, providing fallback if the installed setuptools version
+    doesn't support required features.
+    """
+
+    def __init__(self, name_string, version_string=None):
+        self._plain_distribution = False
+        self._distribution_name = name_string
+        if name_string.startswith('file:') or _is_vcs_url(name_string):
+            return
+
+        self._plain_distribution = True
+        if version_string:
+            version_string = version_string.lstrip()
+            separator = '==' if version_string[0].isdigit() else ' '
+            name_string = separator.join((name_string, version_string))
+        self._requirement = Requirement.parse(name_string)
```

Comment:
if this method terminates in line 278 there will be no `self._requirement` in class instance, which would break things.

Reply:
To access `_requirement` it will first check `_plain_distribution`, so it's ok.

Reply:
It's inconsistency, so it's not ok.

Reply:
I agree with @webknjaz, set self._requirement = None if you have to... just so that it is set in both code paths.